src/Sdk/RequestAdapter.php

```diff
@@ -0,0 +1,236 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\SdkBlueprint\Sdk;
+
+use EoneoPay\Utils\Str;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\ValidationException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestMethodInterface;
+use LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestObjectInterface;
+use LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestOptionAwareInterface;
+use LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestSerializationGroupAwareInterface;
+use LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestValidationGroupAwareInterface;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class RequestAdapter
+{
+    /**
+     * HTTP request method.
+     *
+     * @var string
+     */
+    private $httpMethod;
+
+    /**
+     * The Request method.
+     *
+     * @var string
+     */
+    private $requestMethod;
+
+    /**
+     * The Request object instance.
+     *
+     * @var \LoyaltyCorp\SdkBlueprint\Sdk\Interfaces\RequestObjectInterface
+     */
+    private $object;
+
+    /**
+     * The Serializer instance.
+     *
+     * @var \Symfony\Component\Serializer\Serializer
```

Comment:
|`line 84`|
|---|
The default value of a parameter should always be `null` and you handle the default value within the method.